src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -679,7 +679,17 @@ private async Task<bool> EnumerateArtifact(IEnumeratedArtifact artifact, TContex
                 return false;
             }
 
-            string extension = Path.GetExtension(filePath);
+            string extension;
+            try
+            {
+                extension = Path.GetExtension(filePath);
+            }
+            catch (ArgumentException)
```

Comment:
>ArgumentException [](http://example.com/codeflow?start=19&length=17)

please find a fix that doesn't require catching an exception. i.e., scan the file path to see whether it contains invalid chars.

Reply:
Since the exception is rare, I thought the performance cost of always evaluating the string for illegal characters would outweigh the cost of processing the exception.  I will change it if you think it is better.